lib/nodejs/lib/thrift/http_connection.js

```diff
@@ -214,6 +214,18 @@ HttpConnection.prototype.write = function(data) {
   var req = (self.https) ?
       https.request(self.nodeOptions, self.responseCallback) :
       http.request(self.nodeOptions, self.responseCallback);
+
+  //support for timeout
+  var timeout = self.nodeOptions.timeout;
+  if(timeout){
+    req.on('socket', function (socket) {
+        socket.setTimeout(timeout);  
+        socket.on('timeout', function() {
+            req.abort();
+        });
+    });
+  }
```

Comment:
Reading the doc, simply putting 'timeout' key value to `nodeOptions` (without any code change) would activate request timeout.
`socket.setTimeout` is for idle connections which is very different thing.
So reusing the former value to the latter does not seem a good idea.
Do you need idle timeout for your use-case at all ?

Reply:
It looks like additional work is needed here; @guptanishank will you be moving this forward?

Reply:
I am working on a project which is using node 4 version. I need timeout. Are you guys going to merge this PR ? If not what are possible way to handle timeout thing without touching thrift module?

Reply:
Based on the comments from @nsuke and @bananer, it looks like the solution is not quite right - that's why it hasn't been merged.  Someone needs to step up and finish or make a convincing argument that this is correct.

Reply:
@ramukg are you using this patch?  Does it work?  Have you tested the timeout?